src/qcodes/utils/delaykeyboardinterrupt.py

```diff
@@ -16,8 +16,9 @@ class DelayedKeyboardInterrupt:
 
     Inspired by https://stackoverflow.com/questions/842557/how-to-prevent-a-block-of-code-from-being-interrupted-by-keyboardinterrupt-in-py
     """
-    signal_received = None
-    old_handler = None
+
+    signal_received: Optional[tuple[int, FrameType | None]] = None
+    old_handler: Optional[signal._HANDLER] = None
```

Comment:
I don't really like using a private type here but I don't know what the alternative is (use Any?)